exercises/all-your-base/example.jl

```diff
@@ -0,0 +1,39 @@
+# Implement number base conversion
+
+function all_your_base(digits::Vector, input_base::Integer, output_base::Integer)
+    # error checking behavior
+    if input_base < 2
+        throw(ArgumentError("input base must be >= 2"))
+    end
+
+    if output_base < 2
+        throw(ArgumentError("output base must be >= 2"))
+    end
+
+    # short-circuit for [0] input; doesn't matter the base
+    if isempty(digits)
+        return [0]
+    end
+
+    # encode the list to a single number
+    total::Integer = 0

```

Comment:
```suggestion
    total = 0
```

type inference will handle this